firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallableReference.kt

```diff
@@ -0,0 +1,159 @@
+    this.url = url
+    this.options = options
+  }
+
+  /**
+   * Executes this Callable HTTPS trigger asynchronously.
+   *
+   *
+   * The data passed into the trigger can be any of the following types:
+   *
+   *
+   *  * Any primitive type, including null, int, long, float, and boolean.
+   *  * [String]
+   *  * [List&amp;lt;?&amp;gt;][java.util.List], where the contained objects are also one of these
+   * types.
+   *  * [Map&amp;lt;String, ?&amp;gt;&gt;][java.util.Map], where the values are also one of these
+   * types.
+   *  * [org.json.JSONArray]
+   *  * [org.json.JSONObject]
+   *  * [org.json.JSONObject.NULL]
+   *
+   *
+   *
+   * If the returned task fails, the Exception will be one of the following types:
+   *
+   *
+   *  * [java.io.IOException] - if the HTTPS request failed to connect.
+   *  * [FirebaseFunctionsException] - if the request connected, but the function returned
+   * an error.
+   *
+   *
+   *
+   * The request to the Cloud Functions backend made by this method automatically includes a
+   * Firebase Instance ID token to identify the app instance. If a user is logged in with Firebase
+   * Auth, an auth token for the user will also be automatically included.
+   *
+   *
+   * Firebase Instance ID sends data to the Firebase backend periodically to collect information
+   * regarding the app instance. To stop this, see [ ][com.google.firebase.iid.FirebaseInstanceId.deleteInstanceId]. It will resume with a new
+   * Instance ID the next time you call this method.
+   *
+   * @param data Parameters to pass to the trigger.
+   * @return A Task that will be completed when the HTTPS request has completed.
+   * @see org.json.JSONArray
+   *
+   * @see org.json.JSONObject
+   *
+   * @see java.io.IOException
+   *
+   * @see FirebaseFunctionsException
+   */
+  fun call(data: Any?): Task<HttpsCallableResult?> {
+    return if (name != null) {
+      functionsClient.call(name, data, options)
+    } else {
+      functionsClient.call(url!!, data, options)
+    }
+  }
+
+  /**
+   * Executes this HTTPS endpoint asynchronously without arguments.
+   *
+   *
+   * The request to the Cloud Functions backend made by this method automatically includes a
+   * Firebase Instance ID token to identify the app instance. If a user is logged in with Firebase
+   * Auth, an auth token for the user will also be automatically included.
+   *
+   *
+   * Firebase Instance ID sends data to the Firebase backend periodically to collect information
+   * regarding the app instance. To stop this, see [ ][com.google.firebase.iid.FirebaseInstanceId.deleteInstanceId]. It will resume with a new
+   * Instance ID the next time you call this method.
+   *
+   * @return A Task that will be completed when the HTTPS request has completed.
+   */
+  fun call(): Task<HttpsCallableResult?> {
+    return if (name != null) {
+      functionsClient.call(name, null, options)
+    } else {
+      functionsClient.call(url!!, null, options)
+    }
+  }
```

Comment:
We could probably have a single function with default values here annotated with `@JvmOverloads` to let the compiler generate the other function with no args.

```
  @JvmOverloads
  fun call(data: Any? = null): Task<HttpsCallableResult?> {
    return if (name != null) {
      functionsClient.call(name, data, options)
    } else {
      functionsClient.call(url!!, data, options)
    }
  }
```

Also, do we want to have a `suspend` function version of this?

Reply:
`@JvmOverloads` would be perfect here- but it would also change the docs, since (as it currently stands) they have different docs.